### PR TITLE
gdal: 3.7.0 -> 3.7.1 and add package tests

### PR DESCRIPTION
**Pull request by imincik**
###### Description of changes

Update GDAL to the latest bug fix release and add package tests.

List of GDAL changes: 
https://github.com/OSGeo/gdal/blob/v3.7.1/NEWS.md

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/gdal/default.nix [hunk 1]
```diff
@@ -61,15 +63,15 @@
 , zstd
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: rec {
```
[the remark this answers is not in the document]
Comment (imincik) [reply pkgs/development/libraries/gdal/default.nix]: Good point @SuperSandro2000 . Should be fixed now. Thank you.

**Comment by risicle** [comment]
What's the reason for the backport label?

**Comment by risicle** [comment]
Also it's not evaluating.

**Comment by imincik** [comment]
> Also it's not evaluating.

Thanks, should be fixed now.

**Comment by imincik** [comment]
> What's the reason for the backport label?

It is bug fix release and it meets [backporting criteria](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#criteria-for-backporting-changes).

**Comment by kirillrdy** [comment]
> > What's the reason for the backport label?
> 
> It is bug fix release and it meets [backporting criteria](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#criteria-for-backporting-changes).

as far as I can tell 3.7.0 was not backported, so backporting 3.7.1 only makes sense if 3.7.0 is already in 23.05

File: pkgs/development/libraries/gdal/tests.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+let
+  inherit (gdal) pname version;
+
+in
+runCommand "${pname}-tests" {
+  nativeBuildInputs = [ gdal ];
+  meta.timeout = 60;
+} ''
+    # test version
```
Comment (l0b0) [inline pkgs/development/libraries/gdal/tests.nix:11]: Is there an implicit `errexit` here, or should it be set?

**Comment by imincik** [comment]
> > > What's the reason for the backport label?
> > 
> > 
> > It is bug fix release and it meets [backporting criteria](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#criteria-for-backporting-changes).
> 
> as far as I can tell 3.7.0 was not backported, so backporting 3.7.1 only makes sense if 3.7.0 is already in 23.05

Unfortunately you are right. Thank you. I apologize @risicle . I somehow forgot that 3.7 haven't made it to 23.05.

**Comment by imincik** [comment]
Thanks everybody involved in review, good comments and merging !

File: pkgs/development/libraries/gdal/tests.nix (hunk 2 above)
Comment (l0b0) [reply pkgs/development/libraries/gdal/tests.nix:11]: That's not the same thing though. From `man bash`:

> inherit_errexit
>        If set, command substitution **inherits the value of the errexit option,** instead  of  unsetting
>        it in the subshell environment.  This option is enabled when posix mode is enabled.

**Pull request merged**